pybossa/cache/projects.py

```diff
@@ -125,13 +125,32 @@ def format_task(row, lock_users=[]):
 
     if filter_user_prefs:
         # construct task list for worker view
+
+        # TODO: may need to refactor code once Task Browse page optimization is settled
+        sql = """ SELECT task.id,
+                (
+                    SELECT CAST(COUNT(id) AS FLOAT) as ct FROM task_run
```

Comment:
It looks to me cast to float is redundant.

Reply:
the float value is needed to compute complete percentage for filters and sorting - https://github.com/bloomberg/pybossa/blob/e3114b3f719575aec9920cacfba3ebc9e619ca5e/pybossa/cache/task_browse_helpers.py#L196

Reply:
Not 100% sure how this line was used ``` 'pcomplete': '(coalesce(ct, 0)/task.n_answers)',```, but nevertheless, you don't need to cast to float to get a decimal number.

Also, "coalesce" seems redundant to me as well. The "ct" you get from the SQL will only return 0 or some number (correct me if I'm wrong).

Reply:
Looks like task list does not need decimal number for ct, CAST can be removed.
However, removing coalesce() gives wrong sorting results on task browse page.